webapp/src/components/Modal/BidConfirmationModal/BidConfirmationModal.js

```diff
@@ -24,6 +32,60 @@ export default class BidConfirmationModal extends React.PureComponent {
     onAuthorize: PropTypes.func
   }
 
+  constructor(props) {
+    super(props)
+    this.state = {
+      isLoading: false,
+      balance: null
+    }
+  }
+
+  componentWillMount() {
+    const { token, address } = this.props
+    this.fetchBalance(token, address)
+  }
+
+  componentWillReceiveProps({ token, address }) {
+    if (this.props.token !== token || this.props.address !== address) {
+      this.fetchBalance(token, address)
+    }
+  }
+
+  fetchBalance = async (symbol, address) => {
+    this.setState({ balance: null, isLoading: true })
+
+    try {
+      const contractAddress = TOKEN_ADDRESSES[symbol]
+      const response = await axios.get(
+        `https://api.etherscan.io/api?module=account&action=tokenbalance&contractaddress=${contractAddress}&address=${address}&tag=latest&apikey=${env.get(
+          'REACT_APP_ETHERSCAN_API_KEY'
+        )}`
+      )
+      const balance =
+        Number(response.data.result) / (symbol === 'ZIL' ? 10 ** 12 : 10 ** 18)
+      if (isNaN(balance)) {
+        throw new Error('Error fetching balance')
+      }
+      this.setState({
+        balance,
+        isLoading: false
+      })
+    } catch (error) {
+      window.Rollbar.info('Fetching balance via Infura')
+      const contractName = token.getContractNameBySymbol(symbol)
+      const contract = eth.getContract(contractName)
+      const balance = await contract.balanceOf(address)
```

Comment:
Should we need to convert from _wei_ like `balane / (symbol === 'ZIL' ? 10 ** 12 : 10 ** 18)`

We could have `let balance` and format it after the catch

Reply:
done 👍